messybrainz/db/tests/test_recording.py

```diff
@@ -0,0 +1,149 @@
+        }
+        submit_listens([recording])
+        mbid = recording["recording_mbid"]
+        with db.engine.begin() as connection:
+            msid = data.get_id_from_recording(connection, recording)
+            result = get_recording_cluster_id_using_recording_mbid(connection, mbid)
+            self.assertIsNone(result)
+            link_recording_mbid_to_recording_msid(connection, msid, mbid)
+            result = get_recording_cluster_id_using_recording_mbid(connection, mbid)
+            self.assertEqual(msid, result)
+
+
+    def test_insert_recording_cluster(self):
+        """Tests if clusters are inserted properly into recording_cluster table."""
+
+        recording_1 = {
+            "artist": "Jay‐Z & Beyoncé",
+            "title": "'03 Bonnie & Clyde",
+            "recording_mbid": "5465ca86-3881-4349-81b2-6efbd3a59451"
+        }
+        submit_listens([recording_1])
+
+        recording_2 = {
+            "artist": "Jay‐Z and Beyoncé",
+            "title": "'03 Bonnie & Clyde",
+            "recording_mbid": "5465ca86-3881-4349-81b2-6efbd3a59451"
+        }
+        submit_listens([recording_2])
+
+        with db.engine.begin() as connection:
+            recording_gids = fetch_gids_for_recording_mbid(connection, '5465ca86-3881-4349-81b2-6efbd3a59451')
+            insert_recording_cluster(connection, recording_gids[0], recording_gids)
+            recording_gids = fetch_gids_for_recording_mbid(connection, '5465ca86-3881-4349-81b2-6efbd3a59451')
+            self.assertEqual(len(recording_gids), 0)
+
+
+    def test_create_recording_clusters(self):
+        """Tests the create_recording_clusters to make sure clusters are created properly."""
+
+        msb_listens = self._load_test_data('data_for_creating_recording_cluster.json')
+        submit_listens(msb_listens)
+
+        # Create clusters with empty recording_cluster, recording_redirect tables.
+        clusters_modified, clusters_add_to_redirect = create_recording_clusters()
+        self.assertEqual(clusters_modified, 4)
+        self.assertEqual(clusters_add_to_redirect, 4)
+
+        # Create clusters again to make sure duplication of data does not take place.
+        clusters_modified, clusters_add_to_redirect = create_recording_clusters()
+        self.assertEqual(clusters_modified, 0)
+        self.assertEqual(clusters_add_to_redirect, 0)
+
+        # Add new recordings and create clusters again.
+        recording_1 = {
+            "artist": "Memphis Minnie",
+            "title": "Banana Man Blues",
+            "recording_mbid": "e1efdbee-2904-437f-b0e2-dbb4906b86d2",
+        }
+
+        recording_2 = {
+            "artist": "Jay-Z & Beyonce",
+            "title": "'03 Bonnie & Clyde",
+            "recording_mbid": "5465ca86-3881-4349-81b2-6efbd3a59451"
+        }
+        submit_listens([recording_1, recording_2])
+
+        clusters_modified, clusters_add_to_redirect = create_recording_clusters()
```

Comment:
Just mention here that `recording_1` is a completely new recording while `recording_2` should be clustered with a new one.

Reply:
done, a few lines above